cmd/validate/image.go

```diff
@@ -453,6 +457,61 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
 				return err
 			}
 
+			if data.vsaEnabled {
+				// For each validated component, generate and write a VSA
+				vsaPkgOpts := vsa.Options{
+					OutputDir:      "./", // TODO: Make configurable or use temp dir
+					SigningKeyPath: data.vsaSigningKey,
+				}
+				for _, comp := range components {
+					// VSA generation
+					log.Debugf("[VSA] Generating predicate for image: %s", comp.ContainerImage)
+					pred, err := vsa.GeneratePredicate(cmd.Context(), report, comp, vsaPkgOpts)
+					if err != nil {
+						log.Errorf("[VSA] Failed to generate predicate for image %s: %v", comp.ContainerImage, err)
+						continue
+					}
+					log.Debugf("[VSA] Predicate generated for image: %s", comp.ContainerImage)
+
+					vsaPath := fmt.Sprintf("%s.vsa.json", comp.ContainerImage) // TODO: sanitize filename
+					log.Debugf("[VSA] Writing VSA to %s", vsaPath)
+					err = vsa.WriteVSA(pred, vsaPath)
+					if err != nil {
+						log.Errorf("[VSA] Failed to write VSA for image %s: %v", comp.ContainerImage, err)
+						continue
+					}
+					log.Debugf("[VSA] VSA written to %s", vsaPath)
+
+					if data.vsaSigningKey != "" {
```

Comment:
consider using an early return here
```suggestion
					if data.vsaSigningKey == "" {
						log.Debugf("[VSA] No VSA signing key provided, skipping signing and upload")
						continue
					}
```